### PR TITLE
OPENEUROPA-2910: Add Poetry translation request bundle.

### PR DESCRIPTION
**Pull request by imanoleguskiza**
### Description

Add Poetry translation request bundle.

### Change log

- Added: Add Poetry translation request bundle.
- Changed:
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: modules/oe_translation_poetry/config/install/core.entity_form_display.oe_translation_request.poetry_translation_request.default.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    region: content
+  oe_translation_deadline:
+    weight: 5
+    settings: {  }
+    third_party_settings: {  }
+    type: datetime_default
+    region: content
+hidden:
+  oe_poetry_request_id: true
+  fiel  d_update_of: true
```
Comment (22Alexandra) [inline modules/oe_translation_poetry/config/install/core.entity_form_display.oe_translation_request.poetry_translation_request.default.yml:40]: This is a field that you probably renamed and it should be replaced by "oe_update_of".

Comment (22Alexandra) [reply modules/oe_translation_poetry/config/install/core.entity_form_display.oe_translation_request.poetry_translation_request.default.yml:40]: image

File: modules/oe_translation_poetry/config/install/field.storage.oe_translation_request.oe_contact_information.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    - oe_translation_poetry
+id: oe_translation_request.oe_contact_information
+field_name: oe_contact_information
+entity_type: oe_translation_request
+type: oe_translation_poetry_personal_contact
+settings: {  }
+module: oe_translation_poetry
+locked: false
+cardinality: 1
+translatable: true
```
Comment (22Alexandra) [inline modules/oe_translation_poetry/config/install/field.storage.oe_translation_request.oe_contact_information.yml:15]: Do we need the field storages translatable? Are we supposed to translate the information from these fields?

File: modules/oe_translation_poetry/tests/Kernel/PoetryTranslationRequestTest.php [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      'author' => 'Author',
+      'requester' => 'Requester',
+    ];
+    $poetry_translation_request = $translation_request_storage->create([
+      'bundle' => 'poetry_translation_request',
+      'oe_poetry_request_id' => $poetry_id,
+      'oe_contact_information' => $personal_contact_info,
+      'oe_organisation_information' => $organisation_info,
+      'oe_update_of' => $referenced_request->id(),
+      'oe_translation_deadline' => '10-12-2020',
```
Comment (22Alexandra) [inline modules/oe_translation_poetry/tests/Kernel/PoetryTranslationRequestTest.php:71]: Can you set the current date and then check the default value is +14 days?

File: modules/oe_translation_poetry/tests/Kernel/PoetryTranslationRequestTest.php [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    $translation_request_storage->resetCache();
+    $poetry_translation_request = $translation_request_storage->load($poetry_translation_request->id());
+    $this->assertEqual($poetry_translation_request->bundle(), 'poetry_translation_request');
+    $this->assertEqual($poetry_translation_request->get('oe_poetry_request_id')->first()->getValue(), $poetry_id);
+    $this->assertEqual($poetry_translation_request->get('oe_contact_information')->first()->getValue(), $personal_contact_info);
+    $this->assertEqual($poetry_translation_request->get('oe_organisation_information')->first()->getValue(), $organisation_info);
+    $this->assertEqual($poetry_translation_request->get('oe_update_of')->first()->getValue()['target_id'], $referenced_request->id());
+    $this->assertEqual($poetry_translation_request->get('oe_translation_deadline')->first()->getValue()['value'], '10-12-2020');
+  }
+
```
Comment (22Alexandra) [inline modules/oe_translation_poetry/tests/Kernel/PoetryTranslationRequestTest.php:83]: I would add functional js test to test the form.

File: modules/oe_translation_poetry/config/install/core.entity_form_display.oe_translation_request.poetry_translation_request.default.yml (hunk 1 above)
Comment (imanoleguskiza) [reply modules/oe_translation_poetry/config/install/core.entity_form_display.oe_translation_request.poetry_translation_request.default.yml]: Fixed.

File: modules/oe_translation_poetry/config/install/field.storage.oe_translation_request.oe_contact_information.yml (hunk 2 above)
Comment (imanoleguskiza) [reply modules/oe_translation_poetry/config/install/field.storage.oe_translation_request.oe_contact_information.yml]: Indeed they probably shouldn't be, I removed that.

File: modules/oe_translation_poetry/tests/Kernel/PoetryTranslationRequestTest.php (hunk 3 above)
Comment (imanoleguskiza) [reply modules/oe_translation_poetry/tests/Kernel/PoetryTranslationRequestTest.php:71]: I added that part of the test on the functional test.

File: modules/oe_translation_poetry/tests/Kernel/PoetryTranslationRequestTest.php (hunk 4 above)
Comment (imanoleguskiza) [reply modules/oe_translation_poetry/tests/Kernel/PoetryTranslationRequestTest.php:83]: Done.

File: modules/oe_translation_poetry/config/install/oe_translation.oe_translation_request_type.poetry_translation_request.yml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+langcode: en
+status: true
+dependencies: {  }
+id: poetry_translation_request
```
Comment (upchuk) [inline modules/oe_translation_poetry/config/install/oe_translation.oe_translation_request_type.poetry_translation_request.yml:4]: The entity type is called `oe_translation_request`. Why are we calling the bundle `poetry_translation_request`? Is `poetry` not enough?

**Pull request merged**